microservice/src/main/java/com/skytouch/controller/ProductController.java

```diff
@@ -0,0 +1,40 @@
+package com.skytouch.controller;
+
+
+import com.skytouch.common.model.Product;
+import com.skytouch.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+
+@RestController
+public class ProductController {
+
+    private final ProductService productService;
+
+    @Autowired
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @PutMapping(name = "/insertProduct")
+    @ResponseBody
+    public Long insertProduct(
+            @RequestParam("product") Product product) {
+        return productService.addProduct(product);
+    }
+
+
+    @GetMapping(value = "/products")
+    public List<Product> getProducts() {
```

Comment:
Do you know Spring HATEOAS Resource implementation?
Why not using this instead of just returning the Products List?

Reply:
I read about HATEOAS and i think is a good implementation to get more information from the response, but i created this controller just because I thought the broker would consume this, but is not necessary. So, this controller is eliminated